src/components/Calculator.js

```diff
@@ -1,52 +1,71 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.computeFunction = this.computeFunction.bind(this);
+  }
+
+  computeFunction = (e) => {
+    this.setState((now) => calculate({
+      total: now.total,
+      next: now.next,
+      operation: now.operation,
+    }, e.target.name));
   }
 
   render() {
+    // Destructuring this.state
+    const { total, next, operation } = this.state;
     return (
       <div className="panel">
         <table>
           <thead>
             <tr className="result row">
-              <td colSpan={4}>0</td>
+              <td colSpan={4}>
+                {total}
+                {operation}
+                {next}
+              </td>
             </tr>
           </thead>
           <tbody>
             <tr className="row">
-              <td>AC</td>
-              <td>+/-</td>
-              <td>%</td>
+              <td><button type="button" onClick={this.computeFunction} name="AC">AC</button></td>
+              <td><button type="button" onClick={this.computeFunction} name="+/-">+/-</button></td>
+              <td><button type="button" onClick={this.computeFunction} name="&#37;">%</button></td>
               <td className="orange">
-                <span className="semiC">:</span>
-                <span className="divide">-</span>
+                <button type="button" onClick={this.computeFunction} name="&#247;">&#247;</button>
               </td>
             </tr>
             <tr className="row">
-              <td>7</td>
-              <td>8</td>
-              <td>9</td>
-              <td className="orange">x</td>
+              <td><button type="button" onClick={this.computeFunction} name="7">7</button></td>
+              <td><button type="button" onClick={this.computeFunction} name="8">8</button></td>
+              <td><button type="button" onClick={this.computeFunction} name="9">9</button></td>
+              <td className="orange"><button type="button" onClick={this.computeFunction} name="x">x</button></td>
             </tr>
             <tr className="row">
-              <td>6</td>
-              <td>5</td>
-              <td>4</td>
-              <td className="orange">-</td>
+              <td><button type="button" onClick={this.computeFunction} name="6">6</button></td>
+              <td><button type="button" onClick={this.computeFunction} name="5">5</button></td>
+              <td><button type="button" onClick={this.computeFunction} name="4">4</button></td>
+              <td className="orange"><button type="button" onClick={this.computeFunction} name="-">-</button></td>
             </tr>
```

Comment:
- [ ] Great job so far, one of the best practices is to write DRY code, it will be better to create a Button component and pass the keys to that component to keep our code tidy, it will also help as practice using props and lifecycle methods 👍.

Reply:
Thank you, that was an oversight. It's actually an unused portion of the code. As you can see, I already created my button component.

I will effect that as soon as I can.